python/src/aac/context/language_error.py

```diff
@@ -8,3 +8,23 @@ class LanguageError(Exception):
     """A base class representing a language error condition."""
     message: str = attrib(validator=validators.instance_of(str))
     location: Optional[str] = attrib(validator=validators.optional(validators.instance_of(str)))
+
+    def __init__(self, message: str, location: str):
+        """
+        An init constructor for the Language Error Class.
+
+        Args:
+            message(str): The error message.
+            location(str): The source location of the error.
+        """
+        self.message = message
+        self.location = location
+
+    def __str__(self) -> str:
```

Comment:
While this isn't an incorrect way to set this, if we are wanting it to be a way of fetching a formatted string of the LanguageError content, you would not want to house it in a "protected" method like this with the name mangling. You would have it as a public method without the name mangling (so no surrounding underscores). That way people would understand it is meant to be used and is not just some internal representation. If you are intending for it to only be used by internal logic and not an exposed usable method for end users, then you would make it "private" and only have a preceding underscore to the start of the name, i.e. _get_language_error_as_str. So, we just need to decide how this is wanting to be used and adjust the method name accordingly.

Reply:
__str__ is a special python method, like __init__.  It wont be called.  __str__ is basically whats called when you print() something, it just formats the class into a readable format.

Reply:
Correct. So, this was not added with the intent of being used elsewhere within the code, but just to leverage that functionality?

Reply:
Yes.